pavelib/database.py

```diff
@@ -1,104 +1,126 @@
+    """
+    Update the MYSQL database for bokchoy testing:
+    * Determine if your current cache files are up to date
+      with all the migrations
+    * If not then check if there is a copy up at s3
+    * If so then download then extract it
+    * Otherwise apply migrations as usual
+    """
+    fingerprint = fingerprint_bokchoy_db_files(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+    if does_fingerprint_on_disk_match(fingerprint):
+        print ("DB cache files match the current migrations.")
+        apply_migrations(BOKCHOY_DB_FILES, update_cache_files=False)
+
+    elif is_fingerprint_in_bucket(fingerprint, CACHE_BUCKET_NAME):
+        print ("Found updated bokchoy db files at S3.")
+        refresh_bokchoy_db_cache_from_s3(fingerprint=fingerprint)
+        apply_migrations(BOKCHOY_DB_FILES, update_cache_files=False)
+        compute_fingerprint_and_write_to_disk(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+    else:
+        msg = "{} {} {}".format(
+            "Did not find updated bokchoy db files at S3.",
+            "Loading the bokchoy db files from disk",
+            "and running migrations."
+        )
+        print (msg)
+        apply_migrations(BOKCHOY_DB_FILES, update_cache_files=True)
+        compute_fingerprint_and_write_to_disk(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+
+@needs('pavelib.prereqs.install_prereqs')
+@PassthroughTask
+@timed
+def refresh_bokchoy_db_cache_from_s3(fingerprint=None):
     """
-    Determine if the current state of the bokchoy databases and related files
-    have changed since the last time they were updated in the repository by
-    comparing their fingerprint to the fingerprint saved in the repo
+    If the cache files for the current fingerprint exist
+    in s3 then replace what you have on disk with those.
+    If no copy exists on s3 then continue without error.
     """
-    try:
-        fingerprint_filepath = '{}/common/test/db_cache/bokchoy_migrations.sha1'.format(Env.REPO_ROOT)
-        with open(fingerprint_filepath, 'r') as fingerprint_file:
-            cached_fingerprint = fingerprint_file.read().strip()
-    except IOError:
-        return False
-    current_fingerprint = fingerprint_bokchoy_db_files()
-    return current_fingerprint == cached_fingerprint
+    if not fingerprint:
+        fingerprint = fingerprint_bokchoy_db_files(MIGRATION_OUTPUT_FILES, ALL_DB_FILES)
+
+    bucket_name = CACHE_BUCKET_NAME
+    path = CACHE_FOLDER
+    if is_fingerprint_in_bucket(fingerprint, bucket_name):
+        zipfile_name = '{}.tar.gz'.format(fingerprint)
+        get_file_from_s3(bucket_name, zipfile_name, path)
+
+        zipfile_path = os.path.join(path, zipfile_name)
+        print ("Extracting db cache files.")
+        extract_files_from_zip(BOKCHOY_DB_FILES, zipfile_path, path)
+        os.remove(zipfile_path)
 
 
-def fingerprint_bokchoy_db_files():
+@needs('pavelib.prereqs.install_prereqs')
+@PassthroughTask
+@timed
+def upload_db_cache_to_s3():
```

Comment:
How do we intend on this command getting run? IIRC, there was something we wanted to do like:
* recreate/update/apply migrations to the DB
* check one final time to see if the fingerprint file existed in s3
* if not, push to s3. If so, no-op

Reply:
Right. Yeah, we still need to wire this through.
I recommend we merge and refactor in the next PR.